### PR TITLE
Completing JSON server setup

### LINKED ISSUE #9: Setup db.json
## User story
As a **developer**,
I want to  **get, save and post data.**  
As a beginning, start implement JSON Server for a quick brainstorming, which kind of data I want to collect and give back to the user. 

## Notes/Links
Install JSON-Server.
`npx json-server --watch db.json --port 5000`

Use placeholder code.
```JSON
{
  "posts": [{ "id": 1, "title": "json-server", "author": "typicode" }],
  "comments": [{ "id": 1, "body": "some comment", "postId": 1 }],
  "profile": { "name": "typicode" }
}
```

## Acceptance criteria
- [x] Install JSON-Server
- [x] Setup collections
- [x] Implement data manually

### LINKED ISSUE #13: Integrate JSON-Server and Express Server
## User story
As a **developer**,
I want the load **JSON-Server in an Express middleware.**  

## Notes/Links
https://github.com/typicode/json-server#module
https://www.npmjs.com/package/json-server#module

## Acceptance criteria
- [x] Done
- [x] Integrate middleware

### PR DESCRIPTION
**Pull request by ReneSchmitz**
### closed Issue/s
close #9 
close #13

### CODE REVIEW

File: client/src/utils/api.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  const result = await fetch(`http://localhost:3001/api/groceries`);
+  const groceriesHeader = result.json();
+  return groceriesHeader;
+}
+
+export async function getGroceries() {
+  const result = await fetch(`http://localhost:3001/api/groceries`);
+  const groceries = result.json();
+  return groceries;
+}
```
Comment (AlexisRoe) [inline client/src/utils/api.js:11]: you have to wait for the response to come back, before transforming it to and js object
```suggestion
export async function getGroceries() {
  const response = await fetch(`http://localhost:3001/api/groceries`);
  const data = await response.json();
  return data;
}
```

File: client/src/utils/api.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export async function getGroceriesHeader() {
+  const result = await fetch(`http://localhost:3001/api/groceries`);
+  const groceriesHeader = result.json();
+  return groceriesHeader;
+}
```
Comment (AlexisRoe) [inline client/src/utils/api.js:5]: over all , why there are two different functions, doing exact the same thing?

```suggestion
```

File: client/src/components/GroceriesList.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+export default function GroceriesList() {
+  const [grocery, setGrocery] = useState(null);
+
+  useEffect(() => {
+    async function doGet() {
+      const grocery = await getGroceries();
+      setGrocery(grocery);
+      console.log(grocery);
+      console.log(grocery.title, grocery.name);
```
Comment (AlexisRoe) [inline client/src/components/GroceriesList.js:12]: don't leave them there before uploading it 😉 or maybe make a comment at least, why they have to be there
```suggestion
```

**Review by AlexisRoe — COMMENTED** [review]
Good job

File: client/src/utils/api.js (hunk 1 above)
Comment (ReneSchmitz) [reply client/src/utils/api.js:11]: Check out the project from our OMR's:
https://github.com/miriamhaenle/omr-fetch-demo/blob/main/src/api/fetchData.js

1. Awaiting the `response.json()` is not necessary. 
2. However it doesn't help with the issue to render the content of my db.json onto my page.

File: client/src/utils/api.js (hunk 2 above)
Comment (ReneSchmitz) [reply client/src/utils/api.js:5]: This is a preparation for further steps. Function is not called yet.

File: client/src/components/GroceriesList.js (hunk 3 above)
Comment (ReneSchmitz) [reply client/src/components/GroceriesList.js:12]: Currently they show me that my `GroceriesList` function is working till that particular point. 
Output is:
```(4) [{…}, {…}, {…}, {…}]
0: {title: "fruits", name: Array(3)}
1: {title: "drinks", name: Array(5)}
2: {title: "vegetables", name: Array(7)}
3: {title: "cereals", name: Array(2)}
length: 4
__proto__: Array(0)
```
```
undefined undefined
```

File: client/src/utils/api.js (hunk 1 above)
Comment (AlexisRoe) [reply client/src/utils/api.js:11]: sorry, just common practice 🤷‍♂️

File: client/src/utils/api.js (hunk 2 above)
Comment (AlexisRoe) [reply client/src/utils/api.js:5]: alright, maybe leave a comment for that, so other who review your code, will know

File: client/src/components/GroceriesList.js (hunk 3 above)
Comment (AlexisRoe) [reply client/src/components/GroceriesList.js:12]: okay line 11 is the first one?

if so, then grocery isnt an object, it is an array, so line 12 cant work

File: client/src/components/GroceriesList.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    doGet();
+  }, []);
+
+  return (
+    <>{grocery && grocery.title?.map((fruits) => <div key={fruits}></div>)}</>
```
Comment (AlexisRoe) [inline client/src/components/GroceriesList.js:18]: for example
```suggestion
    <>{grocery && grocery.map((foodItem) => <div key={foodItem.title}>{foodItem.title}</div>)}</>
```

**Review by AlexisRoe — COMMENTED** [review]
😜

File: client/src/components/GroceriesList.js (hunk 4 above)
Comment (ReneSchmitz) [reply client/src/components/GroceriesList.js:18]: 🤯

THX a lot!

File: server.js [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
+const jsonServer = require("json-server");
+const router = jsonServer.router("db.json");
+const middlewares = jsonServer.defaults();
 
 const app = express();
 const port = process.env.PORT || 3001;
 
+app.use(middlewares);
+app.use("/api", router);
+
```
Comment (Sina-Sche) [inline server.js:12]: You should move this below the other app.use expressions, so in your case: move it to line - at least that solved the problem for me ;)

Comment (ReneSchmitz) [reply server.js:12]: Thx it works!

**Pull request merged**